compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -3116,12 +3116,27 @@ fn add_lld_args(
 
     let self_contained_linker = self_contained_cli || self_contained_target;
     if self_contained_linker && !sess.opts.cg.link_self_contained.is_linker_disabled() {
+        // We tell `cc` where to find the lld wrappers, in a similar way to how we locate
+        // codegen-backends dylibs, in the tools search paths, which contain:
+        // - the sysroot's `rustlib` bin path,
+        // - and as a fallback for cases using `--sysroot` without tools: in the default host
+        //   sysroot where rustc is currently located.
+        let mut linker_path_exists = false;
         for path in sess.get_tools_search_paths(false) {
-            cmd.arg({
-                let mut arg = OsString::from("-B");
-                arg.push(path.join("gcc-ld"));
-                arg
-            });
+            let linker_path = path.join("gcc-ld");
+            if linker_path.exists() {
+                linker_path_exists = true;
+                cmd.arg({
+                    let mut arg = OsString::from("-B");
+                    arg.push(linker_path);
```

Comment:
Nit: the push can be done unconditionally, then the logic will be simplified to `linker_path_exists &= linker_path.exists()`.

Reply:
Done